UnityWeld_Editor/AnimatorParameterBindingEditor.cs

```diff
@@ -0,0 +1,337 @@
+                {
+                    // Get rid of old adapter options if we changed the type of the adapter.
+                    if (newValue != targetScript.ViewAdapterTypeName)
+                    {
+                        Undo.RecordObject(targetScript, "Set view adapter options");
+                        targetScript.ViewAdapterOptions = null;
+                    }
+
+                    UpdateProperty(
+                        updatedValue => targetScript.ViewAdapterTypeName = updatedValue,
+                        targetScript.ViewAdapterTypeName,
+                        newValue,
+                        "Set view adapter"
+                    );
+                }
+            );
+
+            Type adapterType;
+            viewAdapterOptionsFade.target = ShouldShowAdapterOptions(targetScript.ViewAdapterTypeName, out adapterType);
+
+            EditorStyles.label.fontStyle = viewAdapterOptionsPrefabModified ? FontStyle.Bold : defaultLabelStyle;
+
+            ShowAdapterOptionsMenu(
+                "View adapter options",
+                adapterType,
+                options => targetScript.ViewAdapterOptions = options,
+                targetScript.ViewAdapterOptions,
+                viewAdapterOptionsFade.faded
+            );
+
+            EditorGUILayout.Space();
+
+            EditorStyles.label.fontStyle = viewModelPropertyPrefabModified ? FontStyle.Bold : defaultLabelStyle;
+
+            var adaptedViewPropertyType = AdaptTypeBackward(viewPropertyType, targetScript.ViewAdapterTypeName);
+            ShowViewModelPropertyMenu(
+                new GUIContent("View-model property", "Property on the view-model to bind to."),
+                TypeResolver.FindBindableProperties(targetScript),
+                updatedValue => targetScript.ViewModelPropertyName = updatedValue,
+                targetScript.ViewModelPropertyName,
+                property => property.PropertyType == adaptedViewPropertyType
+            );
+
+            GUI.enabled = guiPreviouslyEnabled;
+
+            EditorStyles.label.fontStyle = defaultLabelStyle;
+
+            EditorGUILayout.Space();
+
+        }
+
+        private void ShowAnimatorParametersMenu(
+            GUIContent label,
+            Action<AnimatorParameterTypeAndName> propertyValueSetter,
+            AnimatorParameterTypeAndName curPropertyValue,
+            out Type selectedPropertyType
+        )
+        {
+            var animator = targetScript.GetComponent<Animator>();
+
+            AnimatorControllerParameter[] properties = new AnimatorControllerParameter[] { };
+
+            // For whatever reason, accessing Animator.parameters while the GameObject the
+            // Animator is on is inactive causes it to return an empty array, aswell as fill
+            // the unity console with warnings. This behaviour is undocumented, and this is just
+            // a work around which seems to work fine
+            if (!animator.gameObject.activeInHierarchy)
```

Comment:
Maybe the code for getting the properties out of an animator could be refacored out to a separate method. It seems like it could be easily separated from the logic for showing the parameters menu and the method is very large already.

Reply:
you did that so I'm not sure why it didn't remove this comment, but thanks :+1: